src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
-     * SSH connectivity is not reachable.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is not Ready,
+    // the processFulfilledOrder() method must change the OrderState to Failed by adding in that
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceNotReachable() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.READY);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
         mockCloudConnectorFactory(orderInstance);
 
+        // exercise
         Mockito.when(this.sshConnectivity
                 .checkSSHConnectivity(Mockito.any(SshTunnelConnectionData.class)))
                 .thenReturn(false);
-        
-        Assert.assertNull(this.failedOrderList.getNext());
-        
+
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
 
-        /** here may be a false positive depending on how long the machine will take to run the test */
-        Thread.sleep(MAX_SLEEP_TIME);
-
-        Assert.assertNull(this.fulfilledOrderList.getNext());
+        /**
+         * here may be a false positive depending on how long the machine will take to run the test
+         */
+        this.thread.sleep(MAX_SLEEP_TIME);
 
+        // verify
         Order test = this.failedOrderList.getNext();
         Assert.assertNotNull(test);
         Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
         Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of a failed localidentity compute instance is definitely changed to
-     * failed.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is Failed,
+    // the processFulfilledOrder() method must change the OrderState to Failed by adding in that
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceFailed() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
```

Comment:
Do we really need to spy this object?

Reply:
No, I will remove the spy from this object.